zerver/tests/test_message_edit.py

```diff
@@ -1351,10 +1351,12 @@ def notify(user_id: int) -> Dict[str, Any]:
                 content=None,
             )
 
-        for muting_user in get_users_muting_topic(stream.id, change_all_topic_name):
+        for user_topic in get_users_with_user_topic_visibility_policy(
+            stream.id, change_all_topic_name
+        ):
             for user in users_to_be_notified:
-                if muting_user.id == user["id"]:
-                    user["muted_topics"] = get_topic_mutes(muting_user)
+                if user_topic.user_profile_id == user["id"]:
+                    user["muted_topics"] = get_topic_mutes(user_topic.user_profile)
                     break
 
```

Comment:
This updates the current test to work with the new code, but does not expand the test to verify the new code itself.
We should also add some test including topics with other visibility policies and then assert that they are moved as expected.
Might be a good idea to convert `lib/topic_is_muted()` into a generic `topic_has_visibility_policy` or something like that in a prep commit, and then expand on the tests.

Reply:
> Might be a good idea to convert lib/topic_is_muted() into a generic topic_has_visibility_policy or something like that in a prep commit

Done (Please check the first commit)

I have created a new test named [`test_edit_unmuted_topic` ](https://github.com/zulip/zulip/pull/24742/commits/305641938727b142f85099255751a5634f253a90#diff-3a0986e298aade053cf2a8360758507bc4fed8c3d1fe341724e6c7fd0dd720a6R1714)